geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSetCommandsFunctionExecutor.java

```diff
@@ -126,4 +130,19 @@ public long zrevrank(RedisKey key, byte[] member) {
   public byte[] zscore(RedisKey key, byte[] member) {
     return stripedExecute(key, () -> getRedisSortedSet(key, true).zscore(member));
   }
+
+  @Override
+  public long zunionstore(RedisKey destinationKey, List<ZKeyWeight> keyWeights,
+      ZAggregator aggregator) {
+    List<RedisKey> keysToLock = keyWeights.stream()
+        .collect(ArrayList::new, (x, y) -> x.add(y.getKey()), ArrayList::addAll);
+    keysToLock.add(destinationKey);
+    return stripedExecute(destinationKey, keysToLock,
+        () -> {
+          getRegionProvider().ensureKeyIsLocal(destinationKey);
```

Comment:
consider moving this ensureKeyIsLocal check down into RedisSortedSet.zunionstore or do it at the beginning of this method. Does it need to be done from the stripedExecute?